awc/src/request.rs

```diff
@@ -21,9 +21,9 @@ use crate::frozen::FrozenClientRequest;
 use crate::sender::{PrepForSendingError, RequestSender, SendClientRequest};
 use crate::ClientConfig;
 
-#[cfg(feature = "compress")]
+#[cfg(feature = "__compress")]
 const HTTPS_ENCODING: &str = "br, gzip, deflate";
-#[cfg(not(feature = "compress"))]
+#[cfg(not(feature = "__compress"))]
 const HTTPS_ENCODING: &str = "br";
```

Comment:
I have add `cfg_if!` macro as you suggest.
However I did not add it everywhere.
* in struct definition, it make `cargo fmt` crash
* in `awc/src/sender.rs` to me it make the code less readable and produce a big change. I can update it if you think it is better with cfg_if

Reply:
Unless I'm mistaken, as it's written now enabling _any_ compression flag will signal support for _all_ of them. My suggestion for cfg_if! was to make it easier to enumerate the feature flag combinations to get the correct string here.

Reply:
You are right, I have not read usage of this but just replace `cfg!` calls.

Actually looking at the current code I do not really understand how it may works.
For now there is the current definition
```rust
const HTTPS_ENCODING: &str = {
    cfg_if::cfg_if! {
        if #[cfg(feature = "__compress")] {
            "br, gzip, deflate"
        } else {
            "br"
        }
    }
};
```

And the only usage of `HTTPS_ENCODING` is in `awc/src/request.rs+495`:
```rust
        if slf.response_decompress {
            let https = slf
                .head
                .uri
                .scheme()
                .map(|s| s == &uri::Scheme::HTTPS)
                .unwrap_or(true);

            if https {
                slf = slf.insert_header_if_none((header::ACCEPT_ENCODING, HTTPS_ENCODING));
            } else {
                #[cfg(feature = "__compress")]
                {
                    slf = slf.insert_header_if_none((header::ACCEPT_ENCODING, "gzip, deflate"));
                }
            };
        }
```

Please note, here the only change with previous code is `compress` becoming `__compress`.
Moreover the if statement related to switching HTTP / HTTPS is just weird.

To me code should be something like this:
```rust
        if slf.response_decompress {
            #[cfg(feature = "__compress")]
            {
                let mut encoding = vec![];

                #[cfg(feature = "compress-brotli")]
                encoding.push("br");

                #[cfg(feature = "compress-gzip")]
                {
                    encoding.push("gzip");
                    encoding.push("deflate");
                }

                #[cfg(feature = "compress-zstd")]
                encoding.push("zstd");

                assert!(!encoding.is_empty(), "encoding cannot be empty unless __compress feature has been explictily enabled.");
                slf = slf.insert_header_if_none((header::ACCEPT_ENCODING, encoding.join(",")));
            }
        }
```